xyz-connectors/src/main/java/com/here/xyz/connectors/decryptors/PrivateKeyEventDecryptor.java

```diff
@@ -94,19 +94,30 @@
 
   @Override
   public String decryptAsymmetric(final String encoded) {
+    if (encoded == null || encoded.equals("")) {
+      return encoded;
+    }
+
+    // we cannot decode the secret if no private key is set.
     if (privateKey == null) {
       return encoded;
     }
+
+    String tmp = encoded.substring(2, encoded.length() - 2);
```

Comment:
encode.length() > 1  is asserted here

Reply:
We check now if the string is encrypted. This means it is not null, contains markers at beginning and end, so it is at least 4 chars long.